api/provider/aws/task_list.go

```diff
@@ -0,0 +1,75 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (t *TaskProvider) List() ([]*models.TaskSummary, error) {
+	clusterNames, err := listClusterNames(t.AWS.ECS, t.Config.Instance())
+
+	if err != nil {
+		return nil, err
+	}
+
+	summaries := []*models.TaskSummary{}
+	for _, clusterName := range clusterNames {
+		runningTaskARNs, err := t.listTasksByStatus(clusterName, "RUNNING")
+		stoppedTaskARNs, err := t.listTasksByStatus(clusterName, "STOPPED")
+
+		if err != nil {
+			return nil, err
+		}
+
+		environmentID := delLayer0Prefix(t.Config.Instance(), clusterName)
+
+		for _, taskARN := range append(runningTaskARNs, stoppedTaskARNs...) {
+			summary := models.TaskSummary{
+				TaskID:        *taskARN,
+				EnvironmentID: environmentID,
+			}
+			summaries = append(summaries, &summary)
+		}
+
+	}
+
+	if err := t.populateTaskSummaries(summaries); err != nil {
+		return nil, err
+	}
+
+	return summaries, nil
+
+}
+
+func (t *TaskProvider) listTasksByStatus(clusterName, status string) ([]*string, error) {
+	input := &ecs.ListTasksInput{}
+	input.SetCluster(clusterName)
+	input.SetDesiredStatus(status)
```

Comment:
need to do `input.Validate() `

Reply:
This should be [ListTasksPages](https://github.com/aws/aws-sdk-go/blob/master/service/ecs/api.go#L2264)

Reply:
I'm not sure that Validate() is implemented in the sdk for either ListTasksInput or ListTasksPages?
```
$ grep -r "Validate" . | grep Task | grep ecs
./vendor/github.com/aws/aws-sdk-go/service/ecs/api.go:func (s *DeregisterTaskDefinitionInput) Validate() error {
./vendor/github.com/aws/aws-sdk-go/service/ecs/api.go:func (s *DescribeTaskDefinitionInput) Validate() error {
./vendor/github.com/aws/aws-sdk-go/service/ecs/api.go:func (s *DescribeTasksInput) Validate() error {
./vendor/github.com/aws/aws-sdk-go/service/ecs/api.go:func (s *RegisterTaskDefinitionInput) Validate() error {
./vendor/github.com/aws/aws-sdk-go/service/ecs/api.go:func (s *RunTaskInput) Validate() error {
./vendor/github.com/aws/aws-sdk-go/service/ecs/api.go:func (s *StartTaskInput) Validate() error {
./vendor/github.com/aws/aws-sdk-go/service/ecs/api.go:func (s *StopTaskInput) Validate() error {
./vendor/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_ecs_task_definition.go:
```
Validate() is there for the above providers, but doesn't appear to be there for List* methods. Unless I'm overlooking something else, I don't think I can simply invoke input.Validate().

Reply:
oh weird, nvm then